packages/backend/src/payment-method/ilp/connector/core/test/middleware/telemetry.test.ts

```diff
@@ -0,0 +1,84 @@
+import assert from 'assert'
+import { OutgoingAccount, ZeroCopyIlpPrepare } from '../..'
+import { IncomingAccountFactory, RafikiServicesFactory } from '../../factories'
+import { createTelemetryMiddleware } from '../../middleware/telemetry'
+import { createILPContext } from '../../utils'
+
+import * as telemetry from '../../../../../../telemetry/transaction-amount'
+
+const incomingAccount = IncomingAccountFactory.build({ id: 'alice' })
+
+assert.ok(incomingAccount.id)
+const services = RafikiServicesFactory.build({})
+
+const ctx = createILPContext({
+  services,
+  request: {
+    prepare: {
+      amount: 100n
+    } as unknown as ZeroCopyIlpPrepare,
+    rawPrepare: Buffer.from('')
+  },
+  accounts: {
+    incoming: incomingAccount,
+    outgoing: { asset: { code: 'USD', scale: 2 } } as OutgoingAccount
+  },
+  state: {
+    unfulfillable: false,
+    incomingAccount: {
+      quote: 'exists'
+    }
+  }
+})
+
+jest.mock('../../../../../../telemetry/transaction-amount')
```

Comment:
do we need this one here? since you are already calling `mockImplementation` when creating the spy

Reply:
yes, in order to mock the implementation, the function must be a jest.fn()
the spies no longer work without that mock